block/sync.go

```diff
@@ -60,9 +61,9 @@ func (m *Manager) SettlementSyncLoop(ctx context.Context) error {
 				}
 				// if we have the block locally, we don't need to fetch it from the DA.
 				// it will only happen in case of rollback.
-				err := m.applyLocalBlock(currH)
+				err := m.applyLocalBlock()
```

Comment:
why u removed `currH`? we call `applyLocalBlock` from state loop, not sure why `applyLocalBlock` needs to get the height on its own

Reply:
i had a race condition because applylocalblock was trying to reapply a block that was already applied from blocksync. i changed it so it gets the height after the mutex lock.